### DISCUSSION TITLE
Push Trivy json report to elasticsearch

### DISCUSSION THREAD

**Discussion by manish7992** [aquasecurity/trivy#6026] [Q&A]
### Question

can we able to push the trivy json report to elasticseach from trivy cli ?

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

**Comment by nikpivkin** [comment]
Hi @manish7992 ! 

I think `Elasticsearch` provides ways to upload files that you can use. It could be CLI or SDK. Trivy doesn't have integration with Elasticsearch.

**Comment by itaysk** [comment]
trivy does not, and will not have direct integration with elastic or any other vendor. but we have recently introduced the ability to create output plug-ins, so it should be fairly simple to create a plug-in that integrates trivy with elastic. https://aquasecurity.github.io/trivy/v0.48/docs/advanced/plugins/#output-plugins

**Comment by neu7ron2** [comment]
I used JQ json parser to format the trivy JSON result and save each entry as a single line JSON file. You can then setup Elastic Search to pickup the file using the 'Custom log' integration. This was specifically for the Kubernetes Trivy scan option which didn't support the custom plugings (otherwise I would have made a single line JSON parser)

trivy k8s --report=all  --scanners vuln --disable-node-collector --format json -o /var/log/trivytemp.js

jq -c '.Resources[] | {Name: .Name, Namespace: .Namespace, Kind: .Kind, Results: .Results[]?} | {Name: .Name, Namespace: .Namespace, Kind: .Kind,  Target: .Results.Target, vul: .Results.Vulnerabilities[]?} | {TrivyScanTime: now|strftime("%Y-%m-%dT%H:%M:%SZ"), Name: .Name, Namespace: .Namespace, Kind: .Kind,  Target: .Target, VulnerabilityID: .vul.VulnerabilityID, PkgName: .vul.PkgName, Severity: .vul.Severity, InstalledVersion: .vul.InstalledVersion, PrimaryURL: .vul.PrimaryURL, Title: .vul.Title, InstalledVersion: .vul.InstalledVersion, FixedVersion: .vul.FixedVersion}' /var/log/trivytemp.json > /var/log/trivy_log_report.json
